spark-connector/src/main/java/com/datastrato/gravitino/spark/catalog/GravitinoCatalogManager.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+package com.datastrato.gravitino.spark.catalog;
+
+import com.datastrato.gravitino.Catalog;
+import com.datastrato.gravitino.NameIdentifier;
+import com.datastrato.gravitino.Namespace;
+import com.datastrato.gravitino.client.GravitinoClient;
+import com.datastrato.gravitino.client.GravitinoMetaLake;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableMap;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** GravitinoCatalogManager is used to sync catalogs from Gravitino server. */
+public class GravitinoCatalogManager {
+  private static final Logger LOG = LoggerFactory.getLogger(GravitinoCatalogManager.class);
+  private static GravitinoCatalogManager gravitinoCatalogManager;
+
+  private volatile boolean isClosed = false;
+  private HashMap<String, Catalog> gravitinoCatalogs = new HashMap<>();
+  private String metalakeName;
+  private GravitinoClient gravitinoClient;
+
+  private GravitinoCatalogManager(String gravitinoUri, String metalakeName) {
+    this.metalakeName = metalakeName;
+    this.gravitinoClient = GravitinoClient.builder(gravitinoUri).build();
+  }
+
+  // It's not thread safe, and is expected to be invoked only once.
+  public static GravitinoCatalogManager createGravitinoCatalogManager(
+      String gravitinoUrl, String metalakeName) {
+    if (gravitinoCatalogManager != null) {
+      throw new RuntimeException("should not create duplicate GravitinoCatalogManager");
+    }
+    gravitinoCatalogManager = new GravitinoCatalogManager(gravitinoUrl, metalakeName);
+    return gravitinoCatalogManager;
+  }
+
+  public static GravitinoCatalogManager getGravitinoCatalogManager() {
+    if (gravitinoCatalogManager == null) {
+      throw new RuntimeException("GravitinoCatalogManager has not created yet");
+    }
+    if (gravitinoCatalogManager.isClosed) {
+      throw new RuntimeException("GravitinoCatalogManager " + "is already closed");
+    }
+    return gravitinoCatalogManager;
+  }
```

Comment:
I think you can use the singleton to simplify the code here.

Reply:
The current implement is like `singleton`:
1. `GravitinoDriverPlugin` create `GravitinoCatalogManager` and save `GravitinoCatalogManager` to a static variable.
2. Spark will get `GravitinoCatalogManager` through the static variable when loading spark catalog.

why not using singleton is because when creating `GravitinoCatalogManager`, we must provide some parameters like GravitinoUri, metalake.

Reply:
I think you can simplify the method name to `create` and `get`? `GravitinoCatalogManager#createGravitinoCatalogManager` is duplicated as `GravitinoCatalogManager` appears twice.

Reply:
done